src/SAML2/XML/samlp/StatusDetail.php

```diff
@@ -64,7 +65,11 @@ public function setDetail(?\DOMElement $detail): void
      */
     public static function fromXML(DOMElement $xml): object
     {
-        return new self($xml);
+        Assert::same($xml->tagName, 'samlp:StatusDetail');
+        Assert::same($xml->namespaceURI, Constants::NS_SAMLP);
+
+        /** @psalm-var \DOMElement $xml->childNodes[1] */
+        return new self(new Chunk($xml->childNodes[1]));
```

Comment:
Having a look at the core document of the standard, here we also have a problem with the contents. The good news, in this case, is that the contents are supposed to be XML elements indeed, so we could use an array of `SAML2\XML\Chunk` objects.

Reply:
I read this a but too late, so I took the same approach as AuthnContextDecl.. A later commit reverts this and uses Chunks..